crates/storage/src/lib.rs

```diff
@@ -176,6 +177,17 @@ pub trait StorageBatchMutate<Type: Mappable>: StorageMutate<Type> {
         Type::Key: 'a;
 }
 
+/// Allows getting batches of values from the storage,
+/// which can be faster in certain implementations than
+/// getting values one by one.
+pub trait StorageBatchInspect<Type: Mappable> {
+    /// Get a batch of values associated with the provided keys.
+    fn get_batch<'a>(
+        &'a self,
+        keys: BoxedIter<'a, &'a Type::Key>,
+    ) -> BoxedIter<'a, Result<Option<Type::OwnedValue>>>;
```

Comment:
Why we can't use generic for the iterator here and don't use `StorageBatchInspect` in the `den` context?

Reply:
Also curios, is that possible to return `impl Iterator<Item = Result<Option<Type::OwnedValue>>>`?=D

Reply:
Yeah that's what I wanted to do initially, relating to this discussion https://github.com/FuelLabs/fuel-core/issues/2344#issuecomment-2433432356

The problem is that we need a big refactor to implement it for `StructuredStorage` or `GenericDatabase` because they are currently used in trait objects in several places.

There could be something I'm missing here, but ultimately when we use this in the GraphQL API the database is held through a trait object one way or another. I'd love to hear any suggestions if you have any ideas about how to get around this, but anything I have thought of so far would require refactoring the graphql API to use generics.

I would be happy to do the GraphQL API refactor to allow us to work around the trait objects here, but last time we discussed it my understanding was that getting the multi-get implementation with boxed iterators in place was higher priority and that we should hold off any refactors of the GraphQL API since we have bigger refactors planned for it.

Reply:
I meant something like this https://github.com/FuelLabs/fuel-core/pull/2419/files

Reply:
Interesting, thanks. I'll have a look 👀

Reply:
Ah I see. We don't need the `StorageBatchInspect` constraint on the GraphQL ports, which means that the whole implementation can be with iterators and we only need the boxes at the port traits. Thats muuuuch better, thanks for showing!

Reply:
Ah right but we still need the boxed iterators for the `KeyValueInspect` trait. That's why I missed this. Because I tried there and I tried at the graphql level. But it's nice to skip the boxes for this trait in the middle at least.

Reply:
Merged your proposal.